### ISSUE TITLE
Order not executed because of "Gekko is already in process..." 

### ISSUE THREAD

**Issue by danielm001** [askmike/gekko#2470]
**Note: this is the technical bug tracker, please use other platforms for getting support and starting a (non technical) discussion. See the [getting help page](https://gekko.wizb.it/docs/introduction/getting-help.html) for details.**

**I'm submitting a ...**
[X] bug report
[ ] question about the decisions made in the repository

**Action taken** (what you did)
Run tradebot (CLI), Gekko v0.6.5

**Expected result** (what you hoped would happen)
All orders are executed

**Actual result** (unexpected outcome)
Some orders were not executed because of "Received advice to sell/buy however Gekko is already in the process to sell/buy"

**Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, etc)

I had this issue on multiple gekkos (bitfinex and kraken). I noticed that this happens, when the "[ORDER] summary" is missing in the order before. This could be the same issue as #2412.

```
Gekko v0.6.5 (CLI)
Bitfinex EUR/BTC

//LAST SUCCESSFUL BUY (missing "[ORDER] summary")
2018-08-17 20:07:23 (INFO):     2018-08-17T19:52:00Z NO-ADVICE   price= 5670.85891
2018-08-17 20:22:47 (INFO):     2018-08-17T20:07:00Z *** LONG ***  buy price= 5661.96668942
2018-08-17 20:22:47 (INFO):     2018-08-17T20:07:00Z LONG-ADVICE    price= 5661.96669

2018-08-17 20:22:47 (INFO):     We have new trading advice!
2018-08-17 20:22:47 (INFO):              Position: long
2018-08-17 20:22:47 (INFO):              Market price: 5661.96668942
2018-08-17 20:22:47 (INFO):              Based on market time: 2018-08-17 20:21:00

2018-08-17 20:22:47 (INFO):     Trader Received advice to go long. Buying  BTC
2018-08-17 20:38:12 (INFO):     2018-08-17T20:22:00Z LONG-ADVICE	price= 5658.60000
2018-08-17 20:52:31 (INFO):     2018-08-17T20:37:00Z LONG-ADVICE	price= 5662.10000
2018-08-17 21:09:03 (INFO):     2018-08-17T20:52:00Z LONG-ADVICE	price= 5674.20500
...

//ERROR
2018-08-21 10:22:50 (INFO):     2018-08-21T10:07:00Z NO-ADVICE   price= 5581.50000
2018-08-21 10:37:10 (INFO):     2018-08-21T10:22:00Z *** SHORT ***  sell price= 5593.79377326
2018-08-21 10:37:10 (INFO):     2018-08-21T10:22:00Z SHORT-ADVICE   price= 5593.79377

2018-08-21 10:37:10 (INFO):     We have new trading advice!
2018-08-21 10:37:10 (INFO):              Position: short
2018-08-21 10:37:10 (INFO):              Market price: 5593.79377326
2018-08-21 10:37:10 (INFO):              Based on market time: 2018-08-21 10:36:00

2018-08-21 10:37:10 (INFO):     Received advice to sell however Gekko is already in the process to buy
2018-08-21 10:37:10 (INFO):     Canceling buy order first
2018-08-21 10:52:33 (INFO):     2018-08-21T10:37:00Z SHORT-ADVICE        price= 5592.76775
2018-08-21 11:07:59 (INFO):     2018-08-21T10:52:00Z SHORT-ADVICE        price= 5595.54455
```

**Comment by cstegmann** [comment]
Might also be the same as #2465

**Comment by askmike** [comment]
So you've had this both on bfx and kraken? Because I have no problems on other exchanges, let's start out with bitfinex. Could you run this patch and log results? https://github.com/askmike/gekko/pull/2463

**Comment by askmike** [comment]
Meanwhile I'll introduce more logging around Gekko Broker orders, because there are too many issues right now across a number of exchanges.

**Comment by askmike** [comment]
One note about your logs: please turn on debug, that already spits out a lot more.

**Comment by cstegmann** [comment]
@askmike I just noticed I have this too on Binance. I had an open Buy order that was taking a while to fill and then the next advice advised to go short and it didn't cancel and sell. This led me to have a partial filled (exposed enough I think) but my strategy thought I was short at that point.

Then it didn't buy more on a long signal due to it thinking it was already exposed. I think I was 20% exposed at that point.

Is it possible to relay data to the strategy on whether I am long or short or exposed or not? Can I just use an event trigger on a _portfolioValueChange_ in a strategy?

```
2018-08-21 10:13:59 (DEBUG):    [ORDER] statusChange: MOVING
2018-08-21 10:14:00 (DEBUG):    [ORDER] statusChange: OPEN
2018-08-21 10:14:06 (DEBUG):    Requested ZEC/BTC trade data from Binance ...
2018-08-21 10:14:07 (DEBUG):    Processing 3 new trades. From 2018-08-21 08:14:02 UTC to 2018-08-21 08:14:04 UTC. (a few seconds)
2018-08-21 10:14:07 (DEBUG):
      Current market: bear
      Percent change predicted: -0.6224794568842036
      Current price: 0.020732
      Last buy price: 0.020442
      Current position: long
2018-08-21 10:14:07 (INFO):     Received advice to sell however Gekko is already in the process to buy
2018-08-21 10:14:07 (INFO):     Canceling buy order first
2018-08-21 10:14:07 (DEBUG):    syncing private data
2018-08-21 10:14:07 (DEBUG):    syncing private data
2018-08-21 10:14:13 (INFO):     NOT selling, already no exposure
2018-08-21 10:14:16 (DEBUG):    Requested ZEC/BTC trade data from Binance ...
2018-08-21 10:14:17 (DEBUG):    No new trades.
```

```
2018-08-21 11:20:17 (DEBUG):
      Current market: bear
      Percent change predicted: 0.5567317458516279
      Current price: 0.020581
      Last buy price: 0.020442
      Current position: short
2018-08-21 11:20:17 (INFO):     NOT buying, already exposed
```

**Comment by askmike** [comment]
>     2018-08-21 10:14:07 (INFO):     Received advice to sell however Gekko is already in the process to > buy
>     2018-08-21 10:14:07 (INFO):     Canceling buy order first
>     2018-08-21 10:14:07 (DEBUG):    syncing private data
>     2018-08-21 10:14:07 (DEBUG):    syncing private data
>     2018-08-21 10:14:13 (INFO):     NOT selling, already no exposure

There was a bug here: after syncing Gekko did not reset the exposure, which is why the last line said NOT selling. I have pushed a fix for this (in both develop and the bfx debug branch). This one: https://github.com/askmike/gekko/commit/e12eea0279b2c16b2d3e308c09d77e7b24b2d95e

**Comment by danielm001** [comment]
Hi @askmike, thank you for your work! i switched two of my bitfinex bots (eur/btc and usd/bch) to the bfx debug branch and enabled the debug logs. My strategy does not trade too often but i will post the logs here, when i see smth interesting.

**Comment by danielm001** [comment]
both bfx debug gekkos executed the first order last night. It looks good and i also see the "[ORDER] summary" - therefore i guess the next order will be executed as well. i will post additional logs if an order should fail. 

[2470_bitfinex_eur_btc_bfx_debug.log](https://github.com/askmike/gekko/files/2313096/2470_bitfinex_eur_btc_bfx_debug.log)

**Comment by askmike** [comment]
@danielm001 those logs look good. Note that I haven't changed the code in that branch, I only added more logging. Would love to see logs of it failing, thanks a lot!

**Comment by askmike** [comment]
Assuming this is fixed since e12eea0, if it isn't please create a new issue!

**Comment by danielm001** [comment]
ok. no error so far, i think it‘s fixed as well. if not i‘ll open a new issue. thank a lot!
